### PR TITLE
Handle JCEF is not supported error

### PR DESCRIPTION
**Pull request by asafgabai**
- [x] All [tests](https://github.com/jfrog/jfrog-idea-plugin/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
-----
In Android Studio and older versions of IntelliJ-based IDEs, the plugin was not loaded and the following error was received:
```
JCEF is not supported in this env or failed to initialize
```
The error occurred because in newer versions (2020.2 and later) of most JetBrains IDEs, the boot runtime contains JCEF by default, but it's not the case in Android Studio.
If JCEF is not supported, a message with a suggestion to choose another boot runtime will be shown.

image

In addition, this PR fixes another bug regarding opening links inside the webview, instead of in a new browser window.

### CODE REVIEW

File: src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java [hunk 1]
```diff
@@ -159,34 +151,14 @@ private void setLeftPanelContent(JComponent component) {
...
-        }, browser.getCefBrowser());
+    private CefBrowser initVulnerabilityInfoBrowser() throws Exception {
+        tempDirPath = Files.createTempDirectory("jfrog-idea-plugin");
+        Utils.extractFromResources("/jfrog-ide-webview", tempDirPath);
 
         String pageUri = tempDirPath.resolve("index.html").toFile().toURI().toString();
-        browser.loadURL(pageUri);
+        WebviewService webviewService = WebviewService.getInstance();
+        webviewService.setOnLoadEnd(() -> updateIssueOrLicenseInWebview(selectedIssue));
+        return webviewService.initBrowser(pageUri);
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java:161]: Can we register a disposer here? I couldn't find any place you did it:
```java
Disposer.register(this, webviewService);
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/WebviewService.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            public void onLoadError(CefBrowser browser, CefFrame frame, ErrorCode errorCode, String errorText, String failedUrl) {
+                super.onLoadError(browser, frame, errorCode, errorText, failedUrl);
+                Logger.getInstance().error("An error occurred while opening the issue details view: " + errorText);
+            }
+        });
+    }
+
+    public void setOnLoadEnd(Runnable onLoadEnd) {
+        this.onLoadEnd = onLoadEnd;
+    }
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/WebviewService.java:81]: Looks not in use

**Review by yahavi — CHANGES_REQUESTED** [review]
Important - Collapsed in my IDEA sandbox (not in the latest version)

File: src/main/java/com/jfrog/ide/idea/ui/webview/WebviewService.java (hunk 2 above)
Comment (asafgabai) [reply src/main/java/com/jfrog/ide/idea/ui/webview/WebviewService.java:81]: Used in JFrogLocalToolWindow.initVulnerabilityInfoBrowser().

File: src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java [hunk 3]
```diff
@@ -138,56 +135,67 @@ private void refreshView() {
...
+        // "The plugin uses a component named JCEF that seem to be missing in your IDE."
+        HyperlinkLabel pluginNeedsJcefLabel = new HyperlinkLabel();
+        pluginNeedsJcefLabel.setText("The plugin uses a component named JCEF that seem to be missing in your IDE.");
+        ComponentUtils.addCenteredHyperlinkLabel(jcefNotSupportedPanel, pluginNeedsJcefLabel);
 
-        return ComponentUtils.createUnsupportedPanel(noCredentialsPanel);
+        // "To make JCEF available in your IDE, you’ll need to have the IDE use a different boot runtime."
+        HyperlinkLabel replaceBootRuntimeLabel = new HyperlinkLabel();
+        replaceBootRuntimeLabel.setText("To make JCEF available in your IDE, you’ll need to have the IDE use a different boot runtime.");
+        ComponentUtils.addCenteredHyperlinkLabel(jcefNotSupportedPanel, replaceBootRuntimeLabel);
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java:173]: What's the reason for using HyperlinkLabel when the component is no hyperlink?

File: src/main/java/com/jfrog/ide/idea/ui/JFrogToolWindow.java [hunk 4]
```diff
@@ -19,15 +20,22 @@ public class JFrogToolWindow {
 
     void initToolWindow(@NotNull ToolWindow toolWindow, @NotNull Project project, boolean buildsConfigured) {
         ContentManager contentManager = toolWindow.getContentManager();
-        JFrogLocalToolWindow jfrogLocalContent = new JFrogLocalToolWindow(project);
+        JFrogLocalToolWindow jfrogLocalContent = null;
+        try {
+            jfrogLocalContent = new JFrogLocalToolWindow(project);
+        } catch (Exception e) {
+            Logger.getInstance().error("Local view could not be initialized.", e);
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/JFrogToolWindow.java:27]: This is troublesome. In that case, we may like to get an exception. This log message is nearly invisible.

File: src/main/java/com/jfrog/ide/idea/ui/JFrogToolWindow.java [hunk 5]
```diff
@@ -19,15 +20,22 @@ public class JFrogToolWindow {
...
+        }
         JFrogCiToolWindow jFrogCiContent = new JFrogCiToolWindow(project, buildsConfigured);
         addContent(contentManager, jfrogLocalContent, jFrogCiContent);
     }
 
     private void addContent(ContentManager contentManager, JFrogLocalToolWindow jfrogLocalContent, JFrogCiToolWindow jfrogBuildsContent) {
         ContentFactory contentFactory = ContentFactory.SERVICE.getInstance();
-        Content localContent = contentFactory.createContent(jfrogLocalContent, "Local", false);
-        contentManager.addContent(localContent);
+        if (jfrogLocalContent != null) {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/JFrogToolWindow.java:35]: The local view must always be not null

**Review by yahavi — CHANGES_REQUESTED** [review]
Amazing as usual, @asafgabai.
Could you please add a section in the readme about Android Studio, explaining the issue and the solution?

File: src/main/java/com/jfrog/ide/idea/ui/JFrogToolWindow.java (hunk 4 above)
Comment (asafgabai) [reply src/main/java/com/jfrog/ide/idea/ui/JFrogToolWindow.java]: I added an empty state for this case.

**Pull request merged**